src/gnn_reco/data/sqlite_dataset.py

```diff
@@ -116,20 +119,20 @@ def _create_graph(self, features, truth):
             'corsika': abs_pid > 20,
         }
 
-        targets = np.asarray([
-            labels_dict['neutrino'],
-            labels_dict['muon'],
-            labels_dict['noise'],
-        ])
-     
-        x = torch.from_numpy(np.asarray(features)[1:].astype(np.float32)) 
-        y = torch.from_numpy(targets)
+        x = torch.from_numpy(np.asarray(features)[:,1:].astype(np.float64)) 
+        n_pulses = torch.tensor(len(x), dtype = torch.int32)
         graph = Data(
             x=x,
-            y=y.unsqueeze(0),
-            **labels_dict,
+            edge_index= None
         )
-        
+        graph.n_pulses = n_pulses
+
+        for label in labels_dict.keys():
+            graph[label] = labels_dict[label]    
+
+        for truth_key in ['energy', 'position_x', 'position_y', 'position_z', 'azimuth', 'zenith']:
+            graph[truth_key] = torch.tensor(truth_dict[truth_key], dtype = torch.float64)
```

Comment:
Am happy with all of these changes, but out of curiosity: If these are the reconstruction targets, what is the argument for treating these as custom labels rather than a standard target (`y=...`)? Is it to force named access to the targets, as opposed to integer indexing a generic target tensor?